interpreter/exec/eval.ml

```diff
@@ -659,6 +700,36 @@ let rec step (c : config) : config =
             string_of_reftype rt ^ " but got " ^
             string_of_reftype (type_of_ref r)) @@ e.at]
 
+      | RefCastDesc _rt, Ref (NullRef _) :: vs' ->
+        vs', [Trapping "null descriptor reference" @@ e.at]
+
+      | RefCastDesc (NoNull, _), Ref _desc :: Ref (NullRef _) :: vs' ->
+        vs', [Trapping "descriptor cast failure" @@ e.at]
+
+      | RefCastDesc (Null, _), Ref _desc :: Ref ((NullRef _) as r) :: vs' ->
+        Ref r :: vs', []
+
+      | RefCastDesc rt, Ref desc :: Ref r :: vs' ->
+        (try
+          let desc' = Aggr.read_desc r in
+          if eq_ref desc desc' then
+            Ref r :: vs', []
+          else
+            failwith "not_equal"
+        with
+          Failure _ -> vs', [Trapping "descriptor cast failure" @@ e.at]
```

Comment:
I'd avoid relying on exceptions for regular behaviour as much as possible. Perhaps add a `has_desc` predicate to the Aggr module? Or make `read_desc` return an option?

Reply:
Nice, it's way simpler when it's an option.